File: security_agency/security_agency/report/salary_slip_with_bank_details_1/salary_slip_with_bank_details_1.py
# Copyright (c) 2025, Anurag Sahu and contributors
# For license information, please see license.txt

import frappe

# def execute(filters=None):
#     month = filters.get("month")
#     year = filters.get("year")

#     columns = [
#         {"label": "Employee", "fieldname": "employee", "fieldtype": "Link", "options": "Employee", "width": 120},
#         {"label": "Employee Name", "fieldname": "employee_name", "fieldtype": "Data", "width": 150},
#         {"label": "Bank Account No", "fieldname": "bank_ac_no", "fieldtype": "Data", "width": 150},
#         {"label": "Bank Branch", "fieldname": "custom_bank_branch", "fieldtype": "Data", "width": 150},
#         {"label": "IFSC Code", "fieldname": "ifsc_code", "fieldtype": "Data", "width": 120},
#         {"label": "Net Pay", "fieldname": "net_pay", "fieldtype": "Currency", "width": 120}
#     ]

#     data = frappe.db.sql("""
#         SELECT 
#             ss.employee,
#             ss.employee_name,
#             e.bank_ac_no,
#             e.custom_bank_branch,
#             e.ifsc_code,
#             ss.net_pay
#         FROM 
#             `tabSalary Slip` ss
#         LEFT JOIN 
#             `tabEmployee` e ON ss.employee = e.name
#         WHERE 
#             ss.docstatus = 1
#             AND MONTH(ss.start_date) = MONTH(STR_TO_DATE(%(month)s, '%%M'))
#             AND YEAR(ss.start_date) = %(year)s
#     """, filters, as_dict=True)

#     return columns, data
def execute(filters=None):
    filters = filters or {}

    conditions = ""
    values = {}

    if filters.get("month") and filters.get("year"):
        conditions += " AND MONTH(ss.start_date) = %(month)s AND YEAR(ss.start_date) = %(year)s"
        values.update({
            "month": frappe.utils.getdate("2025-" + filters.month + "-01").month if not str(filters.month).isdigit() else filters.month,
            "year": filters.year
        })

    if filters.get("site"):
        conditions += " AND e.custom_site = %(site)s"
        values["site"] = filters.site

    columns = [
        {"label": "Employee", "fieldname": "employee", "fieldtype": "Link", "options": "Employee", "width": 120},
        {"label": "Employee Name", "fieldname": "employee_name", "fieldtype": "Data", "width": 150},
        {"label": "Bank Account No", "fieldname": "bank_ac_no", "fieldtype": "Data", "width": 150},
        {"label": "Bank Branch", "fieldname": "custom_bank_branch", "fieldtype": "Data", "width": 150},
        {"label": "IFSC Code", "fieldname": "ifsc_code", "fieldtype": "Data", "width": 120},
        {"label": "Site", "fieldname": "custom_site", "fieldtype": "Link", "options": "Site", "width": 120},
        {"label": "Net Pay", "fieldname": "net_pay", "fieldtype": "Currency", "width": 120}
    ]

    data = frappe.db.sql(f"""
        SELECT 
            ss.employee,
            ss.employee_name,
            e.bank_ac_no,
            e.custom_bank_branch,
            e.ifsc_code,
            e.custom_site,
            ss.net_pay
        FROM 
            `tabSalary Slip` ss
        LEFT JOIN 
            `tabEmployee` e ON ss.employee = e.name
        WHERE 
            ss.docstatus = 1 {conditions}
    """, values, as_dict=True)

    return columns, data

File: security_agency/security_agency/report/salary_slip_with_bank_details_1/salary_slip_with_bank_details_1.js
// Copyright (c) 2025, Anurag Sahu and contributors
// For license information, please see license.txt

// frappe.query_reports["Salary Slip with Bank Details 1"] = {
// 	"filters": [
// 		{
// 			"fieldname": "month",
// 			"label": "Month",
// 			"fieldtype": "Select",
// 			"options": [
// 				"January",
// 				"February",
// 				"March",
// 				"April",
// 				"May",
// 				"June",
// 				"July",
// 				"August",
// 				"September",
// 				"October",
// 				"November",
// 				"December"
// 			],
// 			"default": frappe.datetime.str_to_obj(frappe.datetime.nowdate()).toLocaleString('default', { month: 'long' }),
// 			"reqd": 1
// 		},
// 		{
// 			"fieldname": "year",
// 			"label": "Year",
// 			"fieldtype": "Int",
// 			"default": new Date().getFullYear(),
// 			"reqd": 1
// 		}
// 	]
// };
frappe.query_reports["Salary Slip with Bank Details 1"] = {
    "filters": [
        {
            "fieldname": "month",
            "label": __("Month"),
            "fieldtype": "Select",
            "options": "January\nFebruary\nMarch\nApril\nMay\nJune\nJuly\nAugust\nSeptember\nOctober\nNovember\nDecember",
            "default": frappe.datetime.str_to_obj(frappe.datetime.get_today()).toLocaleString('default',{ month:'long' })
        },
        {
            "fieldname": "year",
            "label": __("Year"),
            "fieldtype": "Int",
            "default": new Date().getFullYear()
        },
        {
            "fieldname": "site",
            "label": __("Site"),
            "fieldtype": "Link",
            "options": "Site",
            "bold": 1
        }
    ]
};

